nova_vm/src/ecmascript/scripts_and_modules/script.rs

```diff
@@ -68,10 +68,15 @@ pub struct Script<'ctx, 'host> {
     ///
     /// Field reserved for use by host environments that need to associate
     /// additional information with a script.
-    pub(crate) host_defined: Option<HostDefined<'host>>,
+    pub(crate) host_defined: Option<HostDefined>,
+
+    /// Source text of the script
+    ///
+    /// Parsing a script takes ownership of the text.
+    source_text: String,
```

Comment:
It doesn't matter _that_ much, but `Box<str>` might be a better choice to avoid doing any accidental mutations.